### PR TITLE
Add dev (SCM) versioning to pip install

### PR DESCRIPTION
**Pull request by jdegenstein**
CadQuery installed from pip+git gives a version tag:
```
>>> import cadquery as cq
>>> cq.__version__
'2.2.1.dev8+g6c8030d'
```
This page discusses more about how to set this up:
https://github.com/pypa/setuptools_scm

It would be nice to have in build123d too. This requires PEP518.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -2,12 +2,15 @@
 requires = [
     "setuptools>=42",
     "wheel",
+    "setuptools>=45",
```
Comment (gumyr) [inline pyproject.toml:4]: Should setuptools be listed twice here?

**Review by gumyr — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 1 above)
Comment (jdegenstein) [reply pyproject.toml:4]: No it should not

**Comment by jdegenstein** [comment]
The tests are failing due to missing `_version.py`, I added it but I am not sure where else it is supposed to be.

File: src/build123d/_version.py [hunk 2]
```diff
@@ -0,0 +1 @@
+
```
Comment (gumyr) [inline src/build123d/_version.py]: According to https://stackoverflow.com/questions/458550/standard-way-to-embed-version-into-python-package, there should be a `__version__ = "0.0.1"` here. Could this be why the builds are failing?

**Review by gumyr — CHANGES_REQUESTED** [review]

File: src/build123d/_version.py (hunk 2 above)
Comment (jdegenstein) [reply src/build123d/_version.py]: added __version__ = "0.0.1" as requested, and imported in `init.py`

**Pull request closed without merging**